woocommerce/woocommerce-android
# Using site timezone in order details

Part of: #9887

### Why
This PR continues the work of using the store's time zone on the app.  As discussed here pe5pgL-4bG-p2, besides the stats, we also want to use the store time zone for the order date:
> we will use the store time zone for everything order-related (orders, order notes, etc.)

### Description
This PR updates the order mapper class to use the store's timezone when mapping the order model date fields (date created, date modified, date paid). This way, we will achieve consistency between the dates displayed in the order list and those displayed in the order details. Besides that change, I also removed the EMPTY order object property and turned it into a function receiving the created and modified dates. This way, we can pass those dates into the new order and use the store timezone to generate those dates.

⚠️ Customer notes are not handled in this PR and will be tackled in a different one. 

### Testing instructions
To test these changes, you must set your device to a different time zone than your store. I used a store in the US and set the device time zone to Australia. This will create a time zone difference of one day (Thursday in the store should be Friday on the device)

1. Open the order list screen
2. Tap on add new order (+)
3. Check that the date on top of the order is using the store's time zone
4. Add some products and create the order
5. Check that the date on top of the order is using the store's time zone
6. Tap on collect payment
7. Tap on cash 
8. Refresh the orders list and select the order you just created
9. Check that the paid date is using the store's time zone

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/c2c4c8e1-0f7d-4d77-8851-fd01a43548ca
